src/graphorm/index.ts

```diff
@@ -176,7 +178,15 @@ const obj = new GraphORM({
               .where(`sm."userId" = :userId`, { userId: ctx.userId })
               .andWhere(`sm."sourceId" = "${parentAlias}".id`),
         },
-        transform: nullIfNotLoggedIn,
+        transform: (value: SourceMember, ctx: Context, parent: Source) => {
```

Comment:
This is not the right place to add this field. This field is part of the `SourceMember` object and there are various ways to fetch it. The more accurate and future proof solution would be to add a custom column to the `SourceMember` entity and if the requested user and the logged in are not the same we can return null or empty array

Reply:
As a rule of thumb, if you added the field to the `SourceMember` object in graphql schema, it needs to be add to graphorm to the same entity. In this case you added it only to a field in the `Source` object. It can create discrepancies and unexpected behavior in other queries. This is not future proof as the client might decide at some point to use a different query to get the permissions but it will not be set properly

Reply:
Ohh. Now I understand. Do you think we should introduce a separate query for this instead? Since the content here is not really found in the database. Then we can do query batching for these permissions.

Reply:
Ohh. Now I understand. Thank you for explaining. Do you think we should introduce a separate query for this instead? Since the content here is not really found in the database. Then we can do query batching for these permissions.

Reply:
@sshanzel we can use a similar implementation to custom fields using JS as can be seen here:
https://github.com/dailydotdev/daily-api/pull/1188/files/009720b5f15c6f184b65822240ba99a6f450130d#diff-928042d056cac827c70794ef59920facfe32f158ebf447ff7d2826fcad5195aeL143